### PR TITLE
EVENT-661: Ability to easily add a logo or picture for the event

### PR DESCRIPTION
**Pull request by katefree**

### CODE REVIEW

File: app/scripts/controllers/eventOverview.js [hunk 1]
```diff
@@ -19,6 +21,31 @@ angular
     };
 
     $scope.conference = conference;
+    $scope.imageSrc = '';
+    $scope.selectedImage = '';
+    $scope.includeImageToAllPages = false;
+
+    $scope.saveImage = function() {
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/eventOverview.js]:
```suggestion
    $scope.saveImage = () => {
```

File: app/scripts/controllers/eventOverview.js [hunk 2]
```diff
@@ -19,6 +21,31 @@ angular
...
+      $http({
+        method: 'PUT',
+        url: 'conferences/' + conference.id + '/image',
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/eventOverview.js]:
```suggestion
        url: `conferences/${conference.id}/image`,
```

File: app/scripts/controllers/eventOverview.js [hunk 3]
```diff
@@ -19,6 +21,31 @@ angular
...
+        data: {
+          image: $scope.imageSrc,
+          includeImageToAllPages: $scope.includeImageToAllPages,
+        },
+      }).then(function() {
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/eventOverview.js]:
```suggestion
      }).then(() => {
```

File: app/scripts/directives/readFile.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+angular
+  .module('confRegistrationWebApp')
+  .directive('readFile', function($timeout, $q) {
```
Comment (TheNoodleMoose) [inline app/scripts/directives/readFile.js]: Can you change the functions in this file to arrow functions?

File: app/views/eventOverview.html [hunk 5]
```diff
@@ -135,5 +135,73 @@ <h2 class="page-title border" translate>Links</h2>
...
+
+          <input
+            read-file
+            type="file"
+            accept="image/*"
+            id="fileInput"
+            class="form-control-file"
+            ng-model="selectedImage"
+          />
+          <p></p>
```
Comment (TheNoodleMoose) [inline app/views/eventOverview.html]: Is this empty <p> just for spacing purposes?

File: app/scripts/controllers/eventOverview.js [hunk 6]
```diff
@@ -20,12 +20,23 @@ angular
...
     };
 
+    $scope.resetImage = () => {
+      $scope.imageSrc = conference.image;
+      $scope.includeImageToAllPages = conference.includeImageToAllPages;
+    };
+
     $scope.conference = conference;
-    $scope.imageSrc = '';
+    $scope.imageSrc = conference.image;
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/eventOverview.js]: I assume when an event is created, the image is defaulted to `''` right?

File: app/views/eventOverview.html [hunk 7]
```diff
@@ -135,5 +135,73 @@ <h2 class="page-title border" translate>Links</h2>
...
+          </i>
+
+          <div class="imagePreview">
+            <img ng-src="{{imageSrc}}" class="full" alt="" />
+          </div>
+
+          <div class="checkbox">
+            <label translate
+              ><input type="checkbox" ng-model="includeImageToAllPages" />
+              Include the Image to Every Page
```
Comment (TheNoodleMoose) [inline app/views/eventOverview.html]: Should this say "Include the image on every page"?

File: app/scripts/controllers/eventOverview.js (hunk 1 above)
Comment (katefree) [reply app/scripts/controllers/eventOverview.js]: fixed

File: app/scripts/controllers/eventOverview.js (hunk 2 above)
Comment (katefree) [reply app/scripts/controllers/eventOverview.js]: fixed

File: app/scripts/controllers/eventOverview.js (hunk 3 above)
Comment (katefree) [reply app/scripts/controllers/eventOverview.js]: fixed

File: app/scripts/directives/readFile.js (hunk 4 above)
Comment (katefree) [reply app/scripts/directives/readFile.js]: done

File: app/scripts/controllers/eventOverview.js (hunk 6 above)
Comment (katefree) [reply app/scripts/controllers/eventOverview.js]: That is true, if there is already an imaf]ge for the conference then it should be displayed on overview page

File: app/views/eventOverview.html (hunk 7 above)
Comment (katefree) [reply app/views/eventOverview.html]: fixed

File: app/views/eventOverview.html (hunk 5 above)
Comment (katefree) [reply app/views/eventOverview.html]: removed

File: app/scripts/controllers/eventOverview.js [hunk 8]
```diff
@@ -2,77 +2,79 @@
...
-      $scope.imageSrc = conference.image;
-      $scope.includeImageToAllPages = conference.includeImageToAllPages;
-    };
+      $scope.resetImage = () => {
+        $scope.imageSrc = conference.image;
+        $scope.includeImageToAllPages = conference.includeImageToAllPages;
+      };
 
-    $scope.conference = conference;
-    $scope.imageSrc = conference.image;
```
Comment (TheNoodleMoose) [inline app/scripts/controllers/eventOverview.js]: Did you mean to remove this? I guess if you remove this the resetImage function will set it.

**Pull request merged**